python/lsst/ts/observatory/control/base_tcs.py

```diff
@@ -967,6 +993,117 @@ def ready_to_take_data(self):
+            The time which the coordinate trasformation is intended for. If
+            `None` (default) use current time.
+
+        Returns
+        -------
+        azel : `astropy.coordinates.AltAz`
+            Astropy coordinates with azimuth and elevation.
+        """
+        radec_icrs = ICRS(Angle(ra, unit=u.hourangle), Angle(dec, unit=u.deg))
+
+        if time is None:
+            time = salobj.astropy_time_from_tai_unix(salobj.current_tai())
+
+        time.location = self.location
+
+        coord_frame_azel = AltAz(location=self.location, obstime=time)
+
+        azel = radec_icrs.transform_to(coord_frame_azel)
+
+        return azel
+
+    def radec_from_azel(self, az, el, time=None):
+        """Calculate Ra/Dec in ICRS coordinates from Az/El.
+
+        Parameters
+        ----------
+        ra : `float`, `str` or `astropy.coordinates.Angle`
+            Target RA, either as a float (hour), a sexagesimal string
+            (HH:MM:SS.S or HH MM SS.S) coordinates or
+            `astropy.coordinates.Angle`.
+        dec : `float`, `str` or `astropy.coordinates.Angle`
+            Target Dec, either as a float (deg), a sexagesimal string
+            (DD:MM:SS.S or DD MM SS.S) coordinates or
+            `astropy.coordinates.Angle`.
+        time : `astropy.time.core.Time` or `None`, optional
+            The time which the coordinate trasformation is intended for. If
+            `None` (default) use current time.
+
+        Returns
+        -------
+        radec_icrs : `astropy.coordinates.ICRS`
+            Astropy coordinates with azimuth and elevation.
+        """
+
+        if time is None:
+            time = salobj.astropy_time_from_tai_unix(salobj.current_tai())
+
+        time.location = self.location
+
+        coord_frame_azel = AltAz(
+            alt=Angle(el, unit=u.deg),
+            az=Angle(az, unit=u.deg),
+            location=self.location,
+            obstime=time,
+        )
+
+        radec_icrs = coord_frame_azel.transform_to(ICRS)
+
+        return radec_icrs
+
+    def parallactic_angle(self, ra, dec, time=None):
+        """Return parallactic angle for the given Ra/Dec coordinates.
+
+        Parameters
+        ----------
+        ra : `float`, `str` or `astropy.coordinates.Angle`
+            Target RA, either as a float (hour), a sexagesimal string
+            (HH:MM:SS.S or HH MM SS.S) coordinates or
+            `astropy.coordinates.Angle`.
+        dec : `float`, `str` or `astropy.coordinates.Angle`
+            Target Dec, either as a float (deg), a sexagesimal string
+            (DD:MM:SS.S or DD MM SS.S) coordinates or
+            `astropy.coordinates.Angle`.
+        time : `astropy.time.core.Time` or `None`, optional
+            The time which the coordinate trasformation is intended for. If
+            `None` (default) use current time.
+
+        Returns
+        -------
+        parallactic_angle : `astropy.coordinates.Angle`
+            Parallactic angle.
```

Comment:
Below you name your variable pa_angle which is computed by a function called parallactic_angle (which gets imported from the utils script). Please consider updating the doc string to refer to the pa_angle variable instead. The names of the variable and the function and even the function in which this is all coded is quite confusing :)